Code/Framework/AzToolsFramework/AzToolsFramework/UI/PropertyEditor/ComponentEditor.cpp

```diff
@@ -922,22 +922,21 @@ namespace AzToolsFramework
         return false;
     }
 
-    void ComponentEditor::EnteredComponentMode(const AZStd::vector<AZ::Uuid>& componentModeTypes)
+    ComponentEditor::ComponentModeActiveState ComponentEditor::EnteredComponentMode(const AZStd::vector<AZ::Uuid>& componentModeTypes)
     {
         // disable all component cards not matching the ComponentMode type
         if (AZStd::find(
             componentModeTypes.begin(),
             componentModeTypes.end(), m_componentType) == componentModeTypes.end())
         {
             SetWidgetInteractEnabled(this, false);
+            return ComponentEditor::ComponentModeActiveState::ComponentCardDisabled;
         }
         else
         {
-            if (!componentModeTypes.empty())
-            {
-                // only set the first item to be selected/highlighted
-                SetSelected(componentModeTypes.front() == m_componentType);
-            }
+            // only set the first item to be selected/highlighted
+            SetSelected(componentModeTypes.front() == m_componentType);
+            return ComponentEditor::ComponentModeActiveState::ComponentCardSelected;
         }
     }
```

Comment:
update: I don't think this change is correct  - please can you put the logic back to how it was before with the three returns as I'm pretty certain they will all be hit. We need an extra enum state added to cover this (I'm going to test locally to be 100% sure but I'm pretty confident this is the case)